packages/next-app/src/app/internalmanager/(components)/TokenTable.tsx

```diff
@@ -93,38 +95,113 @@ function TableRow({
     <Table.BodyRow key={token.tokenInfo.address}>
       <Table.BodyCell>
         <div className="flex justify-center items-center">
-          <Image
-            src={
-              tokenLogoUri[
-                token?.tokenInfo?.symbol as keyof typeof tokenLogoUri
-              ] || genericTokenLogo
-            }
-            alt="Token Logo"
-            height={28}
-            width={28}
-            quality={100}
-          />
+          <div className="bg-white rounded-full p-1">
+            <Image
+              src={
+                tokenLogoUri[
+                  token?.tokenInfo?.symbol as keyof typeof tokenLogoUri
+                ] || genericTokenLogo
+              }
+              className="rounded-full"
+              alt="Token Logo"
+              height={28}
+              width={28}
+              quality={100}
+            />
+          </div>
         </div>
       </Table.BodyCell>
-      <Table.BodyCell>{token.tokenInfo.symbol}</Table.BodyCell>
-      <Table.BodyCell>{token.tokenInfo.address}</Table.BodyCell>
+      <Table.BodyCell>
+        {token.tokenInfo.name} ({token.tokenInfo.symbol})
+      </Table.BodyCell>
       <Table.BodyCell>{token.balance}</Table.BodyCell>
       <Table.BodyCell>
+        <div className="flex items-center gap-2">
+          {transactionButtons.map((button) => (
+            <TransactionButton
+              key={button.operation}
+              icon={button.icon}
+              operation={button.operation}
+              network={network}
+              token={token}
+              setToken={setToken}
+              disabled={button?.disabled}
+              userAddress={userAddress}
+              setUserAddress={setUserAddress}
+            />
+          ))}
+        </div>
+      </Table.BodyCell>
+    </Table.BodyRow>
+  );
+}
+
+const transactionButtons = [
+  {
+    icon: <PlusCircledIcon width={22} height={22} />,
+    operation: "deposit",
+    disabled: true,
+  },
+  {
+    icon: <MinusCircledIcon width={22} height={22} />,
+    operation: "withdraw",
+  },
+  {
+    icon: <WidthIcon height={22} width={22} />,
+    operation: "transfer",
+  },
+];
+
+function TransactionButton({
+  network,
+  token,
+  userAddress,
+  setToken,
+  setUserAddress,
+  icon,
+  operation,
+  disabled = false,
+}: {
+  network: string;
+  token: ArrElement<GetDeepProp<InternalBalanceQuery, "userInternalBalances">>;
+  userAddress: `0x${string}`;
+  setToken: React.Dispatch<
+    ArrElement<GetDeepProp<InternalBalanceQuery, "userInternalBalances">>
+  >;
+  setUserAddress: React.Dispatch<`0x${string}`>;
+  icon: React.ReactElement;
+  operation: string;
+  disabled?: boolean;
+}) {
+  return (
+    <>
+      {!disabled ? (
         <Link
-          href={`/internalmanager/${network}/withdraw/${token.tokenInfo.address}`}
+          href={`/internalmanager/${network}/${operation}/${token.tokenInfo.address}`}
```

Comment:
we're passing network as props through 3 components to construct this URL, can't we infer it from the route path?

Reply:
the TokenTable component doesn't have the network on the URL yet. But we will do that on[ this issue](https://linear.app/bleu-llc/issue/BAL-310/nested-route-according-to-chain)